### ISSUE TITLE
Can't save .pcl multiple times after importing sound layer

### ISSUE THREAD

**Issue by scribblemaniac** [pencil2d/pencil#830]
### --Issue Summary--
An error occurs when attempting to save a .pcl file a second (or more) time after importing a sound file. The first save works fine, and as a workaround you can close the file, then open it again and save as many times as you want.

### --Actual Results--
The error dialog contains the following information:
```
FileManager::save
strFileName = <redacted>.pcl
layerCount = 4
layer[0] = Layer[id=1, name=Camera Layer, type=5]
layer[1] = Layer[id=2, name=Vector Layer, type=2]
layer[2] = Layer[id=3, name=Bitmap Layer, type=1]
layer[3] = Layer[id=4, name=Sound Layer, type=4]
- Layer[3] failed to save
Layer::save
strDataFolder = <redacted>.pcl.data
- Keyframe[1] failed to save

Error Display
Title: Internal Error
Description: An internal error occurred while trying to save the file. Some or all of your file may not have saved.

System Info
Pencil version: 0.5.4
Build ABI: x86_64-little_endian-lp64
Kernel: darwin 17.2.0
Product name: macOS High Sierra (10.13)
```

### --Steps to reproduce--
1. Start Pencil2d
2. Create a Sound Layer
3. Import a sound (mp3/wav, doesn't seem to matter)
4. Save the file as a .pcl file
5. Save the file again

### --System Information--
+ Pencil2D Version: 0.6.0

+ Operating System: macOS 10.13.1

+ RAM Size: 16 GB

**Comment by Jose-Moreno** [comment]
@scribblemaniac seems to be similar to the issue reported here https://github.com/pencil2d/pencil/issues/808 and here https://github.com/pencil2d/pencil/issues/802 that @chchwy fixed here https://github.com/pencil2d/pencil/commit/4e3453407e5542ed59c78a41656374b140fbb427 might be worth looking into that to see what's happening with the PCL file extension.

**Comment by scribblemaniac** [comment]
@Jose-Moreno I agree, those do seem related, and could potentially both be caused by this issue. The fact that it works after reload would give the false impression that this bug has been fixed. I think the commit you meant to link was this one 3bb4b2e.

I just tested this on the master branch version, and the issue is still present. In fact because I'm in debug mode, instead of showing an error message it just outright crashes! This happens because of [this QAssert](https://github.com/pencil2d/pencil/blob/master/core_lib/structure/layersound.cpp#L146).
